### ISSUE TITLE
[Bug]: `--build_python_zip` seems to not work in py_binary

### ISSUE THREAD

**Issue by ChangFu2000** [aspect-build/rules_py#236]
### What happened?

When building a py_binary with `--build_python_zip` specified, expected .zip artifact is not generated.

### Version

Development (host) and target OS/architectures:
Ubuntu 20.04

Output of `bazel --version`:
bazel 7.0.0 (bazelisk)

Version of the Aspect rules, or other relevant rules from your
`WORKSPACE` or `MODULE.bazel` file:
aspect_rules_py, version: 0.5.0
rules_python, version: 0.22.0

Language(s) and/or frameworks involved:
Python

### How to reproduce

```starlark
# MODULE.bazel

bazel_dep(name = "aspect_rules_py", version = "0.5.0")

bazel_dep(name = "rules_python", version = "0.22.0", dev_dependency = True)

python = use_extension("@rules_python//python/extensions:python.bzl", "python")
python.toolchain(
    name = "python3",
    configure_coverage_tool = True,
    python_version = "3.11",
)
use_repo(python, "python3_toolchains")

register_toolchains(
    "@python3_toolchains//:all",
)
```

```starlark
# BUILD

load("@aspect_rules_py//py:defs.bzl", "py_binary")

py_binary(
    name = "script",
    srcs = ["script.py"],
    main = "script.py",
)
```

Running with: `bazel build //:script --build_python_zip`
```


### Any other information?

_No response_

**Comment by Cartman75** [comment]
If there is a self executing python file, that would seem like a upgrade and much better then zip.. Others can tell me if I am wrong, but currently, the zip is just the best we got right now, sorta deal. I would much prefer to just execute a file like most other code can do.

**Comment by ChangFu2000** [comment]
@alexeagle Thanks for the reply. We are using a docker environment so both .pyz and .pex satisfy our use case. I'm not faimliar with Bazel implementation details so please follow your decision.

**Comment by mattem** [comment]
Closed via 7a9e4b2, rules_py provides pex as the zip implementation.
